### PR TITLE
Update SortActivityDiagram and Reorganise Implementations in DeveloperGuide

### LINKED ISSUE #174: Fix While Notation in diagram

### PR DESCRIPTION
**Pull request by whitesnowx**
Actions:
* Remove `while()` and appropriate notations to match CS2103 UML standards in `SortActivityDiagram.puml`
* Reorder Implementations in `DeveloperGude.md` to match `UserGuide.md`
* Change subheaders of implementations to standardise naming
* Add `Why sort is implemented this way` 

🖊️ 🐧

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -264,12 +299,27 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+The main operation for the sort feature is the `updateSortedPersonList(Comparator<Person> comparator)` method in the `Model` component.
+The following are some explanations for decisions made in the implementation of the sort feature.
+
+Need for multiple `Comparator` objects:
+This is to keep in view for when other commands or enhancements may need the separate attribute predicates. 
+
+Need for `MultiComparator` object:
+This is to map the 1 or more comparator objects and act as a layer of abstraction where `SortCommmand` does need to know how many attributes are used in sorting.
+This also w
```
Comment (iynixil) [inline docs/DeveloperGuide.md]: clarification: incomplete sentence?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -264,12 +299,27 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+`SortCommmandParser` parsing the `Comparator` objects:
+This is to prevent `SortCommand` from taking on more responsibilities (Separation of Concerns).
+
+#### What designs were considered:
 **Aspect: Determining order of sorting of an attribute:**
 
-* **Current Design:** Get order of sorting from user, prompting for an input in the form of toCompare.
+* **Current Design:** Get order of attributes sorting from user input.
```
Comment (iynixil) [inline docs/DeveloperGuide.md]: suggestion: `Get order of attributes sorting` -> `Get sorting order of attributes` may be clearer?

**Review by iynixil — COMMENTED** [review]
lgtm! some comments

**Comment by Pluiexo** [comment]
Noticed that you reordered that statements, as I was gonna do it too. But you can choose to merge this first, then I will make #249 follow your layout when dealing with the conflicts

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (whitesnowx) [reply docs/DeveloperGuide.md]: Amended

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (whitesnowx) [reply docs/DeveloperGuide.md]: Amended

**Review by iynixil — APPROVED** [review]

**Pull request merged**